tower/src/builder/mod.rs

```diff
@@ -459,6 +459,47 @@ impl<L> ServiceBuilder<L> {
     {
         self.layer.layer(service)
     }
+
+    /// Wrap the async function `F` with the middleware provided by this [`ServiceBuilder`]'s
+    /// [`Layer`]'s, returning a new [`Service`].
+    ///
+    /// This is a convenience for doing [`.service(service_fn(handler_function))`][`service_fn`].
```

Comment:
nit, take it or leave it: maybe something like this is slightly nicer:
```suggestion
    /// This is a convenience method which is equivalent to calling
    /// [`ServiceBuilder::service`] with a [`service_fn`], like this:
    /// ```rust
    /// # use tower::{ServiceBuilder, ServiceFn};
    /// # async fn handler_fn(_: ()) -> Result<(), ()> { Ok(()) }
    /// # let _ = {
    /// ServiceBuilder::new()
    ///     // ...
    ///     .service(service_fn(handler_fn))
    /// # };
```

Reply:
Much nicer!